Review

OLODiman11 opened the pull request:
Fixes #15 

Review existing code for:
- Readability
- Duplication
- Potential bugs
- Testability
- Other suggestions

Open "Files changed" tab and add your comments
Please tag your comment with corresponding issue type (ctrl + b to make text **bold**):
**Readability**: Rename this variable...
**Duplication**: Extract a method...

OLODiman11 committed:
Created bugs.txt

OLODiman11 committed:
Added GUT plugin

OLODiman11 committed:
TetrisShape class + Tests

TetrisShape:
	enum Type:
		I_Block
		J_Block
		L_Block
		O_Block
		S_Block
		T_Block
		Z_Block

	static:
		get_i_block_shape()
		get_j_block_shape()
		get_l_block_shape()
		get_o_block_shape()
		get_s_block_shape()
		get_t_block_shape()
		get_z_block_shape()

	type: Type

	_init(Type)
	get_shape()
	rotate()

OLODiman11 committed:
Cnacged TetrisShapes

+get_rotation() -> Array[Array]
~rotate()

OLODiman11 committed:
Added GridManager + some functions

+add_new_shape(type, row, col)
+get_whole_grid()
+try_rotate_shape() -> bool
+try_move_down() -> bool
+try_move_left() -> bool
+try_move_right() -> bool

OLODiman11 committed:
Script was not saved before last commit. So recommiting after saving the script ;)

OLODiman11 committed:
Added tests for GridManager

OLODiman11 committed:
Tests refactoring

OLODiman11 committed:
Added different codes to shapes so they can be distinguished

OLODiman11 committed:
GridManaged modified

OLODiman11 committed:
GridVisualizer Script

OLODiman11 committed:
GameManager

OLODiman11 committed:
HideLabel

OLODiman11 committed:
Main scene + Input map

OLODiman11 committed:
Working version of tetris

OLODiman11 committed:
TetrisShape.get_actual_size()

OLODiman11 committed:
Got rid of while loops

OLODiman11 committed:
Fall preview

OLODiman11 committed:
Array2i data structure

OLODiman11 committed:
GridConfig resource

OLODiman11 committed:
Fast side move + FPS label

S1rMekr committed:
Merge pull request #2 from OLODiman11/core-mechanic

S1rMekr committed:
Added music, sound and modified scripts for music to work

S1rMekr committed:
Delete scripts/sound_manager.gd

S1rMekr committed:
fixed problems told by OloloDiman11, deleted unnecessary stuff like rotation sound and empty sound_manager, which is now audio_manager

S1rMekr committed:
Merge branch 'music' of https://github.com/OLODiman11/tetris into music

S1rMekr committed:
'test'

S1rMekr committed:
added functions for sound play with piece movement, optimized code

S1rMekr committed:
optimized game_manager

S1rMekr committed:
Merge pull request #4 from OLODiman11/music

Music

S1rMekr committed:
New SFX

Updated loose and drop system

S1rMekr committed:
Merge pull request #5 from OLODiman11/loose

New SFX

S1rMekr committed:
Update bugs.txt

OLODiman11 committed:
Merge pull request #6 from OLODiman11/S1rMekr-patch-1

Found a bug in piece movement

S1rMekr committed:
Main menu

Added a new scene "Menu" with a pretty background and interface.
Added a script for a menu with a localization(Only English and Russian available for now)
Put "Menu" scene as the main scene.

S1rMekr committed:
Optimized translation, added "button" node for SFX

1. Made support for a script to hold a variaty of languages
2. Added a separate node with class to improve sound system of menu buttons

S1rMekr committed:
Fixed bug-m1

Was really hard to fix it

S1rMekr committed:
Merge pull request #8 from OLODiman11/7-bug-m1

Fixed bug-m1

S1rMekr committed:
Created godot-tests.yml

Created a CI action

S1rMekr committed:
Merge pull request #10 from OLODiman11/S1rMekr-patch-1

Created godot-tests.yml

S1rMekr committed:
Update godot-tests.yml

Fixed CI action

S1rMekr committed:
Merge pull request #11 from OLODiman11/S1rMekr-patch-1

Update godot-tests.yml

S1rMekr committed:
Update godot-tests.yml

S1rMekr committed:
Merge pull request #12 from OLODiman11/S1rMekr-patch-2

Update godot-tests.yml

S1rMekr committed:
Loose menu, UX and more

1. Added new menu scene for loose event with "restart" and "exit"
2. Changed game_manager so now it doesn't have unnecessary labels
3. Modified language_pack

S1rMekr committed:
Some fixes to previous commit

Added sounds for loose buttons

S1rMekr committed:
hotfix

S1rMekr committed:
Delete scenes/men350E.tmp

S1rMekr committed:
Delete scenes/button.tscn

S1rMekr committed:
Merge pull request #13 from OLODiman11/localization

OLODiman11 committed:
Added get_rotation_state() to indicate how many rotations away the shape is from the initial position

OLODiman11 committed:
Vizualization reworked

Now current shape is a separate node. It would be easier to make transitions from one place to another that way.

OLODiman11 committed:
Added movement transitions

OLODiman11 committed:
Added particles effect on row deletion

OLODiman11 committed:
Changed particles effect on row deletion

OLODiman11 commented:
Common issues:
- Missing type hinting
- `Array[Array]` instead of `Array[PackedInt32Array]`

S1rMekr commented on scenes/FPSLabel.gd:
Probably we should put these script into main folder with scripts. But also devide scripts fodler by groups of logic.

S1rMekr commented on scenes/shape_previews/preview_box.gd:
Put it into scripts fodler

S1rMekr commented on scenes/shape_previews/shape_preview.gd:
We should put it in the scripts folder

S1rMekr commented on scenes/shape_previews/shape_preview.gd:
```diff
@@ -0,0 +1,32 @@
+@export var config: GridConfig = preload("res://resources/grid_config.tres")
+@export var type: TetrisShape.Type
+
+func _init(type: TetrisShape.Type):
```
I thinklk we should not mention "type:" here, because it mentiones in variable

S1rMekr commented on scripts/audio_manager.gd:
```diff
@@ -0,0 +1,11 @@
+@onready var row_sfx = $RowSFX
+@onready var boost_sfx = $BoostSFX
+@onready var drop_sfx = $DropSFX
+@onready var youmoose_sfx = $Youmoose
```
youmoose should be loose_sfx or deleted

OLODiman11 committed:
Export for android

OLODiman11 committed:
Modified gitignore

Excluded android folder because it seems to be automatically created anyway.
Added recommended gitignore template https://github.com/github/gitignore/blob/main/Godot.gitignore
Removed export_presets.cfg as it is in the recommended gitignore from above and seems to contain credentials that are not safe to publish.

OLODiman11 committed:
Created release.keystore and added it to gitignore

S1rMekr committed:
Merge pull request #17 from OLODiman11/android-export

Exporting the game on Android device

OLODiman11 committed:
Added 1 frame delay before shape is attached to the grid

OLODiman11 committed:
Fixed shadowing warnings

OLODiman11 committed:
Precision narrowing warnings

OLODiman11 committed:
Integer division

OLODiman11 committed:
Unused argument

OLODiman11 committed:
Cannot convert int to string

OLODiman11 committed:
Delete scenes/particles_test.tscn

OLODiman11 committed:
Merge pull request #18 from OLODiman11/grid-attachment-delay

Added 1 frame delay before a shape is attached to the grid

OLODiman11 committed:
Merge pull request #19 from OLODiman11/warnings-fixes

Fixed some errors and warnings

OLODiman11 committed:
Added settings menu scene

OLODiman11 committed:
Volume settings logic

S1rMekr committed:
Merge pull request #20 from OLODiman11/settings

Volume settings

S1rMekr committed:
Merge branch 'develop' into vfx

S1rMekr committed:
Merge pull request #14 from OLODiman11/vfx

Added visual effects

S1rMekr committed:
shape_preview_bugfix

OLODiman11 committed:
Merge pull request #21 from OLODiman11/S1rMekr-patch-1

shape_preview_bugfix

OLODiman11 committed:
Fixed grid not updating

OLODiman11 committed:
Fixed tiles slight misallignment + improved readability

OLODiman11 committed:
Fixed some errors

S1rMekr committed:
Merge pull request #22 from OLODiman11/bugfix

S1rMekr committed:
Main Menu improvements

Updated language_pack
Created Theme for buttons
Created class for sliders

S1rMekr committed:
Main UI

Changed the colours of main scene

S1rMekr committed:
Tween buttons, custom VBox, preview_box

Added tweens for buttons
New VBoxContainer
Changed and centered preview_box

S1rMekr committed:
menu_button update

OLODiman11 committed:
Merge pull request #23 from OLODiman11/s1rmekr

Menu UI, Tweens

S1rMekr committed:
new SFX and BackgroundLoop

S1rMekr committed:
Loose screen

S1rMekr committed:
New UI, music and global translation

S1rMekr committed:
Deleted tween in shape_preview

OLODiman11 committed:
Merge pull request #24 from OLODiman11/s1rmekr

More updates

S1rMekr committed:
Some improvements

-UI
-translation
-globals

OLODiman11 committed:
Merge pull request #25 from OLODiman11/s1rmekr

Some improvements
